ts/session/apis/open_group_api/sogsv3/sogsApiV3.ts

```diff
@@ -0,0 +1,501 @@
+ * If you get null, you most likely need to cancel the processing of whatever you are doing
+ */
+function getSogsConvoOrReturnEarly(serverUrl: string, roomId: string): ConversationModel | null {
+  const convoId = getOpenGroupV2ConversationId(serverUrl, roomId);
+  if (!convoId) {
+    window.log.info(`getSogsConvoOrReturnEarly: convoId not built with ${serverUrl}: ${roomId}`);
+    return null;
+  }
+
+  const foundConvo = getConversationController().get(convoId);
+  if (!foundConvo) {
+    window.log.info('getSogsConvoOrReturnEarly: convo not found: ', convoId);
+    return null;
+  }
+
+  if (!foundConvo.isOpenGroupV2()) {
+    window.log.info('getSogsConvoOrReturnEarly: convo not an opengroup: ', convoId);
+    return null;
+  }
+
+  return foundConvo;
+}
+
+/**
+ *
+ * Handle the pollinfo from the response of a pysogs.
+ * Pollinfos contains the subscriberCount (active users), the read, upload and write things we as a user can do.
+ */
+async function handlePollInfoResponse(
+  statusCode: number,
+  pollInfoResponseBody: {
+    active_users: number;
+    read: boolean;
+    token: string;
+    upload: boolean;
+    write: boolean;
+    details: { admins?: Array<string>; image_id: number };
+  },
+  serverUrl: string,
+  roomIdsStillPolled: Set<string>
+) {
+  if (statusCode !== 200) {
+    window.log.info('handlePollInfoResponse subRequest status code is not 200');
+    return;
+  }
+
+  if (!isObject(pollInfoResponseBody)) {
+    window.log.info('handlePollInfoResponse pollInfoResponseBody is not object');
+    return;
+  }
+
+  const { active_users, read, upload, write, token, details } = pollInfoResponseBody;
+
+  if (!token || !serverUrl) {
+    window.log.info('handlePollInfoResponse token and serverUrl must be set');
+    return;
+  }
+
+  if (!roomIdsStillPolled.has(token)) {
+    window.log.info('handlePollInfoResponse room is no longer polled: ', token);
+    return;
+  }
+
+  const foundConvo = getSogsConvoOrReturnEarly(serverUrl, token);
+  if (!foundConvo) {
+    return; // we already print something in getSogsConvoOrReturnEarly
+  }
+  await foundConvo.setPollInfo({
+    read,
+    write,
+    upload,
+    subscriberCount: active_users,
+    details: pick(details, 'admins', 'image_id'),
+  });
+}
+
+async function filterOutMessagesInvalidSignature(
+  messagesFilteredBlindedIds: Array<OpenGroupMessageV4>
+) {
+  const sentToWorker = messagesFilteredBlindedIds.map(m => {
+    return {
+      sender: PubKey.cast(m.session_id).key, // we need to keep the prefix if this is a blinded or not pubkey
+      base64EncodedSignature: m.signature,
+      base64EncodedData: m.data,
+    };
+  });
+  const startVerify = Date.now();
+  const signatureValidEncodedData = (await callUtilsWorker(
+    'verifyAllSignatures',
+    sentToWorker
+  )) as Array<string>;
+  const signaturesValidMessages = compact(
+    (signatureValidEncodedData || []).map(validData =>
```

Comment:
Why would you map [] here?

Reply:
well, this call is just making 

```ts
const signaturesValidMessages = signatureValidEncodedData ? compact(
    signatureValidEncodedData.map(validData =>
      messagesFilteredBlindedIds.find(m => m.data === validData)
    )
  ) : [];
```

But having the [] map call makes it a little biteasier to read, I think.
Let me know if that works for you